### DISCUSSION TITLE
additionalUserInfo not exposed on Android

### DISCUSSION THREAD

**Discussion by braincomb** [capawesome-team/capacitor-firebase#261] [Q&A]
Hi @robingenz ,

I see that this feature has been added as part of the #30 , however in my testing I'm currently unable to see `additionalUserInfo` being exposed on an Android device. I can see it though when testing on a web browser.

It seems I'm only getting the `credential` from the SignInResult and nothing else. Using Google sign-in.

Any ideas?

**Config:**
```
    "FirebaseAuthentication": {
      "skipNativeAuth": true,
      "providers": ["apple.com", "google.com", "facebook.com"]
    }
```

**Using:**
"@capacitor-firebase/authentication": "^1.2.0"
Android 9 / Chrome Mobile WebView 107

💊   Capacitor Doctor  💊 

Latest Dependencies:

  @capacitor/cli: 4.5.0
  @capacitor/core: 4.5.0
  @capacitor/android: 4.5.0
  @capacitor/ios: 4.5.0

Installed Dependencies:

  @capacitor/core: 4.4.0
  @capacitor/android: 4.4.0
  @capacitor/cli: 4.4.0
  @capacitor/ios: 4.4.0

**Comment by braincomb** [comment]
Update: I found that setting `skipNativeAuth` to `false` fixes the issue and I can see `additionalUserInfo` being returned. This is an issue for me however, since I'm using Firebase JS SDK and have an Apple Sign-In provider so skipNativeAuth=true is required.

**Comment by robingenz** [comment]
You can set `skipNativeAuth` to `true` for Apple only: 
```ts
const appleResult = await FirebaseAuthentication.signInWithApple({ skipNativeAuth: true });
...
const googleResult = await FirebaseAuthentication.signInWithGoogle({ skipNativeAuth: false });
```
This way you should get `additionalUserInfo` for both providers because Apple returns this information with and without `skipNativeAuth`.

**Comment by braincomb** [comment]
Hi @robingenz ,

Thanks for the response, using Google sign-in with `FirebaseAuthentication.signInWithGoogle({ skipNativeAuth: false });` worked great on Android. 

However, I  just tested Apple sign-in on iOS device, and `FirebaseAuthentication.signInWithApple({ skipNativeAuth: true });` does NOT return `additionalUserInfo`. Only changing it to `skipNativeAuth: false` makes it return the data, but since I'm using JS SDK that is not an option for me.

**Reply by robingenz** [reply]
On iOS, unfortunately, the plugin only has access to this data after authentication. I will improve the documentation regarding this. What data do you need exactly?

**Reply by braincomb** [reply]
I'm only looking for `additionalUserInfo.isNewUser`, if that is only available after the authentication, then I'm assuming `isNewUser` will always return false?

I'll probably just end up implementing my own way of checking new user, due to the inconsistent way this is returned.

Thanks again.
